packages/keyring-controller/src/KeyringController.ts

```diff
@@ -219,12 +222,14 @@ export class KeyringController extends BaseController<
     await this.verifySeedPhrase();
 
     this.updateIdentities(newAccounts);
+    let addedAccountAddress = '';
     newAccounts.forEach((selectedAddress: string) => {
       if (!oldAccounts.includes(selectedAddress)) {
-        this.setSelectedAddress(selectedAddress);
+        addedAccountAddress = selectedAddress;
```

Comment:
This variable will change every time `!oldAccounts.includes(selectedAddress)`. Is that what we want?

In addition there don't seem to be any tests that check this logic or check that `addedAccountAddress` is non-empty, can those be added?

Reply:
What do you think about this refactor? [73c539d](https://github.com/MetaMask/core/pull/1095/commits/73c539d9f8b3bdabdf44054e275acb36b71d0c10)

I'll await until I get your input before developing the proper test

Reply:
Yeah that refactor makes more sense!